include/grpc/grpc.h

```diff
@@ -427,6 +436,8 @@ void grpc_channel_watch_connectivity_state(
     completions are sent to 'completion_queue'. 'method' and 'host' need only
     live through the invocation of this function. */
 grpc_call *grpc_channel_create_call(grpc_channel *channel,
+                                    grpc_call *parent_call,
+                                    gpr_uint32 inheritance_mask,
```

Comment:
Thinking about this on the weekend..
I've redefined GRPC_PROPAGATE_DEFAULTS to (effectively) 0xffff - and this is what I'd encourage wrapped languages to use.

It gives us the flexibility to define new contexts that are default propagated or not: if we want existing implementations to propagate, then we define the bit to be in the lower 16, if we want to default to not propagating, we pick the higher 16 bits.